Drafts/1NNN-VLP.md

```diff
@@ -0,0 +1,86 @@
+# New proposed syntax for mixins
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | Victor Porton <porton@narod.ru>                                              |
+| Implementation: | (links to implementation PR if any)                             |
+| Status:         | Will be set by the DIP manager (e.g. "Approved" or "Rejected")  |
+
+## Abstract
+
+The programmer sometimes needs to instantiate mixin templates with the same arguments repeatedly, like:
+
+```
+mixin template A(int x, float y) {
+  // ...
+}
+
+mixin template B(int x, float y) {
+  // ...
+}
+
+I propose to shorten the above code with the following syntax:
+
+template composite(int x, float y) {
+  mixin A {
+    // ...
+  }
+  mixin B {
+    // ...
+  }
+}
+
+// ...
+
+alias ourMixins = composite(1, 2.0)
+
+struct Test {
+  // put two mixins into Test
+  ourMixins.A;
+  ourMixins.B;
+}
+```
+
+### Reference
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+Don't repeat (as in the first example above) the template arguments multiple times.
+It may shorten code, and more importante lessen the errors, as when arguments change,
+we would need to change only in one place of the code, not say two times (for an
+example above).
+
+## Description
+
```

Comment:
> BTW, we can also allow to define mixins not in a template at all

Please be more specific about this. Is this suggested as an optional part of this DIP, or just as a possible future concept for another DIP?

If you meant the former, please include the details.

Reply:
It is an optional part of this DIP. Moreover, it is a strongly advised suggestion in my DIP.

Reply:
Defining mixin blocks as `mixin M {}` instead of `mixin template M() {}` seems like just a minor change in syntax. What's the justification? It also limits future mixin invocation enhancements, as it is very close to `mixin M;`.